lib/puppet_litmus/serverspec.rb

```diff
@@ -1,13 +1,25 @@
 # frozen_string_literal: true
 
-# helper functions for running puppet commands, and helpers
+# helper functions for running puppet commands. They execute a target system specified by ENV['TARGET_HOST']
 module PuppetLitmus::Serverspec
+  # Applies a manifest twice. First checking for errors. Secondly to make sure no changes occur.
+  #
+  # @param manifest [String] puppet manifest code to be applied.
+  # @return [Boolean] The result of the 2 apply manifests.
   def idempotent_apply(manifest)
     manifest_file_location = create_manifest_file(manifest)
     apply_manifest(nil, catch_failures: true, manifest_file_location: manifest_file_location)
     apply_manifest(nil, catch_changes: true, manifest_file_location: manifest_file_location)
   end
 
+  # Applies a manifest. returning the result of that apply. Mimics the apply_manifest from beaker
+  #
+  # @param manifest [String] puppet manifest code to be applied.
+  # @param opts [Hash] Alters the behaviour of the command. Valid options are:
+  # :catch_changes [Boolean] exit status of 1 if there were changes.
```

Comment:
From the rendered version it looks like this isn't caught. Probably needs indenting.